modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerGetCacheSizeTest.java

```diff
@@ -0,0 +1,128 @@
+import org.junit.Test;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.apache.ignite.internal.commandline.CommandHandler.EXIT_CODE_OK;
+import static org.apache.ignite.testframework.GridTestUtils.assertContains;
+import static org.apache.ignite.util.GridCommandHandlerIndexingUtils.CACHE_NAME;
+import static org.apache.ignite.util.GridCommandHandlerIndexingUtils.createAndFillCache;
+import static org.apache.ignite.util.GridCommandHandlerIndexingUtils.GROUP_NAME;
+
+/**
+ * Test for get cache size
+ */
+public class GridCommandHandlerGetCacheSizeTest extends GridCommandHandlerClusterPerMethodAbstractTest {
+    /**
+     *
+     */
+    public static final int GRID_CNT = 2;
+
+    /**
+     * Max mutable cache size. Real size no more than 10000 (hardcoded in createAndFillCache) + MAX_CACHE_SIZE
+     * */
+    public static final int MAX_CACHE_SIZE = 96 * 1024;
+
+    /**
+     * Default cache size in createAndFillCache
+     * */
+    public static final int DEFAULT_CACHE_SIZE = 10000;
+
+    /**
+     * Test new answer from cache list command
+     * */
+    @Test
+    public void testValidateGridCommandHandlerGetCacheSizeTest() throws Exception {
+        IgniteCache<Integer, GridCommandHandlerIndexingUtils.Person> filledCache = null;
+
+        filledCache = fillCache();
+
+        injectTestSystemOut();
+
+        final long testSize = filledCache.sizeLong();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "list", "."));
+
+        String out = testOut.toString();
+
+        assertContains(log, out, "cacheSize=" + testSize);
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "list", ".", "--groups"));
+
+        out = testOut.toString();
+
+        assertContains(log, out, "cacheSize=" + (testSize + DEFAULT_CACHE_SIZE));
+        assertContains(log, out, "cacheSize=" + DEFAULT_CACHE_SIZE);
+    }
+
+    /**
+     * Fill cache with random data
+     * @return IgniteCache
+     */
+    private IgniteCache<Integer, GridCommandHandlerIndexingUtils.Person> fillCache() throws Exception {
+        Ignite ignite = prepareGridForTest();
+
+        IgniteCache<Integer, GridCommandHandlerIndexingUtils.Person> cache = ignite.cache(CACHE_NAME);
+
+        int maxCacheSizeCntr = 0;
+
+        ThreadLocalRandom rnd = ThreadLocalRandom.current();
+
+        while(maxCacheSizeCntr++ < MAX_CACHE_SIZE) {
+            int id = rnd.nextInt(MAX_CACHE_SIZE);
+            cache.put(id, new GridCommandHandlerIndexingUtils.Person(id, "name" + id));
+        }
+
+        return cache;
+    }
+
+    /**
+     * Create and fill nodes.
+     *
+     * @throws Exception
+     */
+    private Ignite prepareGridForTest() throws Exception{
+        Ignite ignite = startGrids(GRID_CNT);
+
+        ignite.cluster().active(true);
+
+        Ignite client = startGrid(CLIENT_NODE_NAME_PREFIX);
+
+        createAndFillCache(client, CACHE_NAME, GROUP_NAME);
+        createAndFillCache(client, CACHE_NAME + "0", GROUP_NAME);
```

Comment:
i prefer grp1 = GROUP_NAME and grp2 = GROUP_NAME + "0" for example, for more readable code, explain guy who first time reads this code, he need to store in his head what all this + "0" means and belongs. It`s all up to you of course, you can store as is and try to proceed with core review.